tensorflow_probability/python/distributions/loglogistic.py

```diff
@@ -0,0 +1,161 @@
+import tensorflow.compat.v2 as tf
+
+from tensorflow_probability.python.bijectors import exp as exp_bijector
+from tensorflow_probability.python.distributions import logistic
+from tensorflow_probability.python.distributions import transformed_distribution
+from tensorflow_probability.python.internal import assert_util
+from tensorflow_probability.python.internal import dtype_util
+from tensorflow_probability.python.internal import tensor_util
+
+__all__ = [
+    'LogLogistic',
+]
+
+
+class LogLogistic(transformed_distribution.TransformedDistribution):
+  """The log-logistic distribution."""
+
+  def __init__(self,
+               loc,
+               scale,
+               validate_args=False,
+               allow_nan_stats=True,
+               name='LogLogistic'):
+    """Construct a log-logistic distribution.
+
+    The LogLogistic distribution models positive-valued random variables
+    whose logarithm is a logistic distribution with loc `loc` and
+    scale `scale`. It is constructed as the exponential
+    transformation of a Logistic distribution.
+
+    Args:
+      loc: Floating-point `Tensor`;
+        the loc of the underlying logistic distribution(s).
+      scale: Floating-point `Tensor`;
+        the scale of the underlying logistic distribution(s).
+      validate_args: Python `bool`, default `False`. Whether to validate input
+        with asserts. If `validate_args` is `False`, and the inputs are
+        invalid, correct behavior is not guaranteed.
+      allow_nan_stats: Python `bool`, default `True`. If `False`, raise an
+        exception if a statistic (e.g. mean/mode/etc...) is undefined for any
+        batch member If `True`, batch members with valid parameters leading to
+        undefined statistics will return NaN for this statistic.
+      name: The name to give Ops created by the initializer.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      super(LogLogistic, self).__init__(
+          distribution=logistic.Logistic(
+            loc=loc,
+            scale=scale,
+            allow_nan_stats=allow_nan_stats
+          ),
+          bijector=exp_bijector.Exp(),
+          validate_args=validate_args,
+          parameters=parameters,
+          name=name)
+
+  @classmethod
+  def _params_event_ndims(cls):
+    return dict(scale=0, concentration=0)
+
+  @property
+  def loc(self):
+    """Distribution parameter of the underlying distribution."""
+    return self.distribution.loc
+
+  @property
+  def scale(self):
+    """Distribution parameter of the underlying distribution"""
+    return self.distribution.scale
+
+  def _mean(self):
+    with tf.control_dependencies(
+        [] if self.allow_nan_stats else [  # pylint: disable=g-long-ternary
+          assert_util.assert_less(
+            self.scale,
+            tf.ones([], dtype=self.dtype),
+            message='Mean undefined for scale > 1.'),
+        ]):
+      mean = tf.math.exp(self.loc) / sinc(self.scale)
```

Comment:
For passing distribution properties test you will need to do something like:

`scale = tf.convert_to_tensor(self.scale)`

and then use scale.

Multiple uses of a `self....` property will fail the test. This will need to be changed here and below